Library/Homebrew/formula.rb

```diff
@@ -922,6 +897,33 @@ def plist_startup
     self.class.plist_startup
   end
 
+  # This method can be used to provide a plist.
+  # @see https://www.unix.com/man-page/all/5/plist/ Apple's plist(5) man page
+  # <pre>plist do
+  #  {
+  #    Label: "#{plist_name}.override",
+  #    ProgramArguments: ["#{opt_bin}/example", "--do-this"],
+  #    RunAtLoad: true,
+  #    KeepAlive: true,
+  #    StandardErrorPath: "/dev/null",
+  #    StandardOutPath: "/dev/null",
+  #  }
+  # end</pre>
+  def plist
+    plists = self.class.plist.map do |block|
+      plist = instance_eval(&block)
+      label = plist[:Label] ||= plist_name
+      [label, plist.to_plist]
```

Comment:
Consider building a hash here instead of turning into a hash later. `plists[plist[:Label]] = plist.to_plist`.